### PR TITLE
Navbar

### PR DESCRIPTION
**Pull request by melatran**
- NavBar tests complete
- Added a current user method in application to change login/logout button to display

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,8 @@
 class ApplicationController < ActionController::Base
+
+  helper_method :current_user
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
```
Comment (perryr16) [inline app/controllers/application_controller.rb:6]: nice, this will be quite useful going forward

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -20,8 +20,12 @@
...
                     <ul class="navbar-nav ml-auto">
                         <li class="nav-item"><a class="nav-link js-scroll-trigger" href="#quiz">Take the Fear Quiz</a></li>
-                        <li class="nav-item"><a class="nav-link js-scroll-trigger" href="#login">Login</a></li>
-                        <li class="nav-item"><a class="nav-link js-scroll-trigger" href="#logout">Logout</a></li>
+                        <% if current_user.nil? %>
+                          <li class="nav-item"><a class="nav-link js-scroll-trigger" href="#login">Login</a></li>
+                          <li class="nav-item"><a class="nav-link js-scroll-trigger" href="#register">Register</a></li>
+                        <% else %>
+                          <li class="nav-item"><a class="nav-link js-scroll-trigger" href="#logout">Logout</a></li>
+                        <% end %>
```
Comment (perryr16) [inline app/views/layouts/application.html.erb:28]: Could we add a few more like game library, profile, home, search

Comment (melatran) [reply app/views/layouts/application.html.erb:28]: Yup, I was thinking those would be like links for a registered user's nav bar only.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -12,16 +12,22 @@
   <body>
   <nav class="navbar navbar-expand-lg navbar-light fixed-top" id="mainNav">
             <div class="container">
-                <a class="navbar-brand js-scroll-trigger" href="#page-top">I Scream Arcade</a>
+                <a class="navbar-brand js-scroll-trigger" href="/">I Scream Arcade</a>
```
Comment (melatran) [inline app/views/layouts/application.html.erb:15]: This part can be changed to a logo later but for now when you click on it, it will go to the homepage

**Pull request merged**